siliconcompiler/schema/schema_obj.py

```diff
@@ -500,7 +578,10 @@ def write_tcl(self, fout, prefix=""):
 
         for key in allkeys:
             typestr = self.get(*key, field='type')
-            value = self.get(*key)
+            if self.get(*key, field='pernode') != 'none':
+                value = self.get(*key, step=step, index=index)
```

Comment:
If a task-specific value has not been set on an 'optional' parameter, will this still return the correct "step=None, index=None" value?